Automattic/jetpack
# Facebook Widget: Fix URL error when adding widget

Fixes #20836 

#### Changes proposed in this Pull Request:
 * Add the keys from the $new_instance array to the top level of the array returned
   by the update method. This allows the widget to be updated from wp-admin/widgets.php.
 * In the widget method, set the $like_args value to the default values, and only set the
   variable to the instance values if they exist. This prevents a PHP notice from being
   generated when the like_args key doesn't exist.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
These changes need to be tested in the following situations:
 * The widget is added/changed on the wp-admin/widgets.php page.
 * The widget is added/changed on the wp-admin/customize.php page.
 * The widget is added/changed using the new widget block editor.
 * The widget is added/changed using the classic widget editor.

##### Set up the test site
1. Install, activate, and connect this branch of Jetpack.
2. Enable the Extra Sidebar Widgets module by navigating to wp-admin -> Jetpack -> Settings -> Writing. Turn on the `Make extra widgets available for use on your site...` setting.

##### Test with the widget block editor
3. Navigate to Appearance -> Widgets.
4. Add a Facebook Page widget and click update. The widget should display properly on both the Widgets page and on the site's front-end.
5. Change the widget's settings (height, width, etc.) and verify that they are updated correctly.
6. Navigate to Appearance -> Customize.
7. Repeat Steps 4 and 5 in the Customizer. The Facebook Page widget should display properly in the Customizer and on the site's front-end.

##### Test with the classic widget editor
8. Install and activate the [Classic Widgets plugin](https://wordpress.org/plugins/classic-widgets/).
9. Repeat steps 3-7 with the classic widgets editor.
